source/server/connection_handler_impl.cc

```diff
@@ -69,71 +60,141 @@ void ConnectionHandlerImpl::ActiveListener::removeConnection(ActiveConnection& c
-          factory, scope, listener_tag) {}
+ConnectionHandlerImpl::ActiveListener::ActiveListener(ConnectionHandlerImpl& parent,
+                                                      Network::ListenerConfig& config)
+    : ActiveListener(parent,
+                     parent.dispatcher_.createListener(config.socket(), *this, config.bindToPort()),
+                     config) {}
 
 ConnectionHandlerImpl::ActiveListener::ActiveListener(ConnectionHandlerImpl& parent,
                                                       Network::ListenerPtr&& listener,
-                                                      Network::FilterChainFactory& factory,
-                                                      Stats::Scope& scope, uint64_t listener_tag)
-    : parent_(parent), factory_(factory), listener_(std::move(listener)),
-      stats_(generateStats(scope)), listener_tag_(listener_tag) {}
+                                                      Network::ListenerConfig& config)
+    : parent_(parent), listener_(std::move(listener)),
+      stats_(generateStats(config.listenerScope())), listener_tag_(config.listenerTag()),
+      config_(config),
+      legacy_stats_(new Filter::Listener::ProxyProtocol::Config(config.listenerScope())) {}
 
 ConnectionHandlerImpl::ActiveListener::~ActiveListener() {
+  while (!sockets_.empty()) {
+    ActiveSocketPtr removed = sockets_.front()->removeFromList(sockets_);
+    parent_.dispatcher_.deferredDelete(std::move(removed));
+  }
+
   while (!connections_.empty()) {
     connections_.front()->connection_->close(Network::ConnectionCloseType::NoFlush);
   }
 
   parent_.dispatcher_.clearDeferredDeleteList();
 }
 
-ConnectionHandlerImpl::SslActiveListener::SslActiveListener(
-    ConnectionHandlerImpl& parent, Ssl::ServerContext& ssl_ctx, Network::ListenSocket& socket,
-    Network::FilterChainFactory& factory, Stats::Scope& scope, uint64_t listener_tag,
-    const Network::ListenerOptions& listener_options)
-    : ActiveListener(parent,
-                     parent.dispatcher_.createSslListener(parent, ssl_ctx, socket, *this, scope,
-                                                          listener_options),
-                     factory, scope, listener_tag) {}
-
 Network::Listener*
 ConnectionHandlerImpl::findListenerByAddress(const Network::Address::Instance& address) {
+  ActiveListener* listener = findActiveListenerByAddress(address);
+  return listener ? listener->listener_.get() : nullptr;
+}
+
+ConnectionHandlerImpl::ActiveListener*
+ConnectionHandlerImpl::findActiveListenerByAddress(const Network::Address::Instance& address) {
   // This is a linear operation, may need to add a map<address, listener> to improve performance.
   // However, linear performance might be adequate since the number of listeners is small.
   // We do not return stopped listeners.
-  auto listener = std::find_if(
+  auto iter = std::find_if(
```

Comment:
nit: IMO `listener` is more clear. if you want to change I would do `listener_it`

Reply:
I found 'listener' confusing. Changed to 'listener_it'